### PR TITLE
bevy_webgl2 render backend

### PR DESCRIPTION
**Pull request by mrk-its**

### CODE REVIEW

**Comment by cart** [comment]
Wow you've been busy :smile:

File: crates/bevy_render/src/pipeline/pipeline.rs [hunk 1]
```diff
@@ -139,10 +145,16 @@ impl PipelineDescriptor {
...
+        let mut layouts = {
+            let mut layouts = vec![vertex_spirv.reflect_layout(bevy_conventions).unwrap()];
+            log::info!("vertex shader layout: {:#?}", layouts[0]);
+            if let Some(ref fragment_spirv) = fragment_spirv {
+                layouts.push(fragment_spirv.reflect_layout(bevy_conventions).unwrap());
+                log::info!("fragment shader layout: {:#?}", layouts[1]);
+            };
+
+            layouts
+        };
```
[the remark this answers is not in the document]
Comment (mrk-its) [reply crates/bevy_render/src/pipeline/pipeline.rs:157]: I've just reverted all changes in this file

**Comment by chemicstry** [comment]
Can this also function as a native OpenGL backend? Afaik, the only difference is wasm bindings.

Amazing work btw 💯

**Comment by mrk-its** [comment]
> Can this also function as a native OpenGL backend? Afaik, the only difference is wasm bindings.
> Something like [glow](https://github.com/grovesNL/glow) can be used to support many platforms at once
Definitely. I'm finishing cleanup of this PR (a lot of mess landed here while exploring how everything works), after that I'll try to explore glow. Thanks!

**Comment by zicklag** [comment]
Hey, I just want to say how glad I am that somebody is working on this! Web and OpenGL support are something I would really like to see in Bevy! :smiley:

Also, I can vouch for `glow` being pretty cool and in my ( somewhat limited ) experience, it makes it very easy to handle GL On Whatever. :grin:

**Comment by kirawi** [comment]
Ah, so cool!

**Comment by mrk-its** [comment]
Closing as bevy_webgl2 is now available as external plugin: https://github.com/mrk-its/bevy_webgl2

**Pull request closed without merging**